src/Mod/Fem/Gui/ViewProviderFemConstraintOnBoundary.cpp

```diff
@@ -107,22 +104,56 @@ void ViewProviderFemConstraintOnBoundary::highlightReferences(const bool on)
             }
         }
         else {
-            if (subSet.second[0].find("Vertex") != std::string::npos &&
-                !originalPointColors[base].empty()) {
+            if (!originalPointColors[base].empty()) {
                 vp->PointColorArray.setValues(originalPointColors[base]);
                 originalPointColors[base].clear();
             }
-            else if (subSet.second[0].find("Edge") != std::string::npos &&
-                     !originalLineColors[base].empty()) {
+            else if (!originalLineColors[base].empty()) {
                 vp->LineColorArray.setValues(originalLineColors[base]);
                 originalLineColors[base].clear();
             }
-            else if (subSet.second[0].find("Face") != std::string::npos &&
-                     !originalFaceColors[base].empty()) {
+            else if (!originalFaceColors[base].empty()) {
                 vp->DiffuseColor.setValues(originalFaceColors[base]);
                 originalFaceColors[base].clear();
             }
         }
     }
+
+    if (subSets.empty()) {
+        // there is nothing selected but previous selection may have highlighting
+        // reset that highlighting here
+        for (auto& ogPair : originalPointColors) {
+            if (ogPair.second.empty()) continue;
+            Part::Feature* base = static_cast<Part::Feature*>(ogPair.first);
```

Comment:
Here no static_cast is needed because the key of the originalPointColors map already is a Part::Feature.

Reply:
Done.